### ISSUE TITLE
[FEA] Support eval mode in dynamicemb

### ISSUE THREAD

**Issue by jiashuy** [NVIDIA/recsys-examples#35]
**Is your feature request related to a problem? Please describe.**
Under eval mode in pytorch, we shouldn't change the weight.
However, HKV may evict keys in the dynamicemb's forward, causing weigth changes.

**Describe the solution you'd like**
Identify the status(model.train() / model.eval()) and select different code paths in dynamicemb.
For eval mode, using `find` and not `find_or_insert` to lookup the HKV.

**Comment by shijieliu** [comment]
in inference, the missed key can be initialized as 0. So we dont need unique in eval
